src/App.js

```diff
@@ -35,16 +35,63 @@ const App = () => {
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
 
+  const clickOnSquares = (squareId) => {
+    console.log('Is this thing working');
+    let newSquares = [...squares];
+    console.log(newSquares);
+    for (let row = 0; row < newSquares.length; row++) {
+      for (let columns = 0; columns < newSquares.length; columns++) {
+        if (newSquares[row][columns].id === squareId) {
+          newSquares[row][columns].value = 'X';
+        }
+      }
+    }
```

Comment:
We also want to make sure we mark squares that haven't been marked yet. We can do so by checking if the value of that square is empty. 

```suggestion
    for (let row = 0; row < newSquares.length; row++) {
      for (let columns = 0; columns < newSquares.length; columns++) {
        if (newSquares[row][columns].id === squareId && newSquares[row][columns].value === '') {
          newSquares[row][columns].value = 'X';
        }
      }
    }
```